src/runtime.rs

```diff
@@ -479,36 +485,17 @@ struct ActiveQuery {
     dependencies: Option<FxIndexSet<DatabaseKeyIndex>>,
 
     /// Stores the entire cycle, if one is found and this query is part of it.
-    cycle: Vec<DatabaseKeyIndex>,
-}
-
-pub(crate) struct ComputedQueryResult<V> {
-    /// Final value produced
-    pub(crate) value: V,
-
-    /// Minimum durability of inputs observed so far.
-    pub(crate) durability: Durability,
-
-    /// Maximum revision of all inputs observed. If we observe an
-    /// untracked read, this will be set to the most recent revision.
-    pub(crate) changed_at: Revision,
-
-    /// Complete set of subqueries that were accessed, or `None` if
-    /// there was an untracked read.
-    pub(crate) dependencies: Option<FxIndexSet<DatabaseKeyIndex>>,
-
-    /// The cycle if one occured while computing this value
-    pub(crate) cycle: Vec<DatabaseKeyIndex>,
+    cycle: Option<Cycle>,
```

Comment:
I wonder it it'd be better to pull `Arc` out and say: `cycle: Option<Arc<Cycle>>`, as that more obviously reflects the setup where all active queries share the same cycle.

Reply:
Hmm. The `Cycle` type is exposed to end-users -- the intent is that they might store it in their recovery information -- so I kind of prefer to present a simple API to them and not expose the underlying `Arc`. Thoughts?